plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -498,7 +498,8 @@ private void invokeNoMoreSplitsIfNecessary()
                         : (directory.getCurrentDirectories().size() > 0 ? directory.getCurrentDirectories().get(0).getPath() : null);
 
                 if (baseOrDeltaPath != null && AcidUtils.OrcAcidVersion.getAcidVersionFromMetaFile(baseOrDeltaPath, fs) < 2) {
-                    throw new TrinoException(NOT_SUPPORTED, "Hive transactional tables are supported with Hive 3.0 and only after a major compaction has been run");
+                    throw new TrinoException(NOT_SUPPORTED, "Hive transactional tables are supported since Hive 3.0." +
+                            " If you have upgraded from Hive before 3, make sure a major compaction has been run at least once after the upgrade.");
```

Comment:
`Hive before 3` -> `older Hive`?

Reply:
So major companions don't need to be run later on? That seems odd?

Reply:
Major compactions are not needed because we read delta files, and minor-compacted (also delta) files without issue.

Major compaction is required for data produced by Hive < 3, since it's effectively a rewrite.

> Hive before 3 -> older Hive?


Will change to "If you have upgraded from an older version of Hive"